qiskit/circuit/library/arithmetic/piecewise_chebyshev.py

```diff
@@ -0,0 +1,327 @@
+
+    @f_x.setter
+    def f_x(self, f_x: Optional[Callable[[int], float]]) -> None:
+        """Set the function to be approximated.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            f_x: The new function to be approximated.
+        """
+        if self._f_x is None or f_x != self._f_x:
+            self._invalidate()
+            self._f_x = f_x
+
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def degree(self) -> int:
+        """The degree of the polynomials.
+
+                Returns:
+                    The degree of the polynomials.
+        """
+        return self._degree
+
+    @degree.setter
+    def degree(self, degree: Optional[int]) -> None:
+        """Set the error tolerance.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            degree: The new degree.
+        """
+        if self._degree is None or degree != self._degree:
+            self._invalidate()
+            self._degree = degree
+
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def breakpoints(self) -> List[int]:
+        """The breakpoints for the piecewise approximation.
+
+                Returns:
+                    The breakpoints for the piecewise approximation.
+        """
+        breakpoints = self._breakpoints
+        n_l = 2 ** self.num_state_qubits
+
+        # If the last breakpoint is < n_l, add the identity polynomial
+        if self._breakpoints[-1] < n_l:
+            # Add n_l as the last breakpoint since that's what the algorithm expects
+            breakpoints = self._breakpoints + [n_l]
+
+        # If the first breakpoint is > 1, add the identity polynomial
+        if self._breakpoints[0] > 0:
+            breakpoints = [0] + self._breakpoints
+
+        return breakpoints
+
+    @breakpoints.setter
+    def breakpoints(self, breakpoints: Optional[List[int]]) -> None:
+        """Set the breakpoints for the piecewise approximation.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            breakpoints: The new breakpoints for the piecewise approximation.
+        """
+        if self._breakpoints is None or breakpoints != self._breakpoints:
+            self._invalidate()
+            self._breakpoints = breakpoints
+
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def polynomials(self) -> List[List[float]]:
+        """The polynomials for the piecewise approximation.
+
+                Returns:
+                    The polynomials for the piecewise approximation.
+        """
+        if self.num_state_qubits is None:
+            return [[]]
+
+        n_l = 2 ** self.num_state_qubits
+
+        # Add n_l as a
+        num_intervals = len(self._breakpoints)
```

Comment:
Should this use `self.breakpoints` instead?

Reply:
changed in the code